src/enum_set.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <bitset>
+
+template<typename T>
+class EnumSet {
+private:
```

Comment:
I think a helpful typedef is warranted here:

    using base_type_t = std::underlying_type<T>::type;

to avoid repeating `std::underlying_type<T>::type`.